### ISSUE TITLE
Different answer with the original train data

### ISSUE THREAD

**Issue by haosdent** [OptimalScale/LMFlow#324]
I use `data/alpaca/train/train_52002.json` to train, and in it, there is a question "How did Julius Caesar die?"
```
    {
      "text": "Input: Instruction: How did Julius Caesar die? \n Output: Julius Caesar was assassinated by a group of up to 60 conspirators, led by Gaius Cassius Longinus and Marcus Junius Brutus, in the Senate House on the Ides of March (15 March) of 44 BC. \n\n"
    }
```

However, after the training finish, I ask the same question to the bot, it return total different result to me and with other buggy outputs.
![image](https://user-images.githubusercontent.com/950759/234859158-cb99af53-2a69-46c8-9ce3-e4357c133ef0.png)

The model I use is `pinkmanlove/llama-7b-hf` and the deepspeed config is `configs/ds_config_zero2.json`.

**Comment by shizhediao** [comment]
seems that the model did not fit the data very well. How about training more epochs with a larger learning rate to force the model overfit the data?

As for the irrelevant output, it could be solved by adding seperator (e.g., <sep>) to the end of the data.

**Comment by haosdent** [comment]
Noted, let me increase epoch to try again.

**Comment by haosdent** [comment]
I change

**Comment by haosdent** [comment]
I changed to `--num_train_epochs 5.0 --learning_rate 1e-3` and removed all other training data. But the answer is still different when I ask the exact same thing to the bot. 
Any other suggestions to fix this? @shizhediao

**Comment by haosdent** [comment]
> what's the training data are you using?

Self-generation Q&A, based on ChatGPT's input.

**Comment by haosdent** [comment]
Is it related to the prompt_structure in `data/alpaca/train/train_52002.json` is different with the format that expected for `scripts/run_finetune_with_lora.sh` and `pinkmanlove/llama-7b-hf` ?

**Comment by shizhediao** [comment]
I think it is reasonable if the prediction is different from the training data (the LLaMA model has a good diversity).
I have the same observation as you when I test a training prompt.

The question need to be solved is the buggy output as you mentioned. We should add seperators to the data. For example, "Input: <INPUT_SENTENCE> ### Output: <OUTPUT_SENTENCE> ###"

**Comment by haosdent** [comment]
Understand, thanks for your patient explanations Orz
